docs/001_develop/03_client-capabilities/014_utility-methods/08_mixins-util.mdx

```diff
@@ -0,0 +1,77 @@
+---
+title: 'Mixins utilities'
+sidebar_label: 'Mixins utilities'
+id: mixins-util
+keywords: [utils, utility, mixins, augment, addition, lifecycle]
+tags:
+  - utils
+  - utility
+  - mixins
+  - augment
+  - addition
+  - lifecycle
+---
+
+The `mixins` category provides specialized functionalities that can be mixed into Genesis elements to enhance their capabilities and facilitate common patterns used within applications.
+
+:::info
+For Genesis code syntax only
+:::
+
+## 1. Lifecycle Mixin
+
+This mixin is designed to augment Genesis elements with additional control over their connect and disconnect lifecycle hooks. It's particularly useful for elements that are part of custom layouts or need to manage their lifecycle events in a granular manner.
+
+## Key Features
+
+- **Conditional Lifecycle Execution:** Offers methods to determine whether connect and disconnect lifecycle callbacks should run based on the element's current state and container.
+- **Layout Integration:** Seamlessly integrates with custom layouts by providing mechanisms to respect the dragging state and lifecycle update tokens.
+- **Cleanup Timeout:** Ensures elements removed from the layout are correctly cleaned up after a predefined timeout, avoiding memory leaks.
+
+## Examples
+
+#### `LifecycleMixin` Usage
+
+```typescript
+import { LifecycleMixin } from '@genesislcap/foundation-utils';
+import {customElement, GenesisElement, observable} from '@genesislcap/web-core';
+
+@customElement({
+  name: 'my-element',
+  template: html`<!-- template here -->`,
+})
+export class MyExampleComponent extends LifecycleMixin(GenesisElement) {
+  // Your element's implementation
```

Comment:
can we flesh this out with a simple example? Not even sure myself what we can do with this.